smolyay/samples.py

```diff
@@ -0,0 +1,415 @@
+        self._num_points_per_level[1:] = [
+            rule(i) - rule((i - 1)) for i in range(1, self.max_level + 1)
+        ]
+
+        self._end_level = numpy.cumsum(self._num_points_per_level)
+        self._start_level = self._end_level - self._num_points_per_level
+        # points
+        num_points = self._end_level[-1]
+        points = numpy.zeros(num_points, dtype=float)
+        for i in range(1, self.max_level + 1):
+            degree = 2**i
+            # find indexes of extrema not already found. Fraction index/degree
+            # cannot be further simplified
+            if i == 1:
+                indexes = numpy.arange(0, degree + 1, 2, dtype=int)
+            else:
+                indexes = numpy.arange(1, degree, 2, dtype=int)
+                divisible_indexes = numpy.gcd(indexes, degree) > 1
+                indexes = indexes[~divisible_indexes]
+            # calculate new points
+            points[self._start_level[i] : self._end_level[i]] = -numpy.cos(
+                numpy.pi * indexes / degree
+            )
+        self._points = self._scale_to_domain(points, [-1, 1])
+
+
+class SlowNestedClenshawCurtisPointSet(NestedClenshawCurtisPointSet):
+    r"""Set for Clenshaw Curtis slow exponential growth.
+
+    The :attr:`points` for this interpolation scheme are the extrema of the
+    Chebyshev polynomials of the first kind on the domain :math:`[-1, 1]`:
+
+    .. math::
+
+        x_i^* = -\cos(\pi i/n), i = 0,...,n
+
+    For the special case :math:`n = 0`, there is only one point :math:`x_0^* = 0`.
+
+    The nested Clenshaw Curtis points come from the nested extrema of the
+    Chebyshev polynomials of the first kind :math:`n` where n is part of a 
+    sequence
+     
+    .. math::
+
+    n = \begin{cases}
+         0 & \text{ if } k = 0\\ 
+         2^{k} & \text{ if } k > 0 
+    \end{cases}
+
+    where k is a whole number. 
+
+    As the extrema of the Chebyshev polynomials of degree :math:`n` are nested,
+    meaning any :math:`n` will have extrema at the same points as every n that
+    precedes it in the sequence. The extrema are organized into levels such that
+    a level L that is nonempty will contain extrema of some degree n that are
+    not found in any preceding n.
+
+    The total number of unique points increases exponentially with increasing n.
+    To limit the rate of increasing points with increasing level L to a linear
+    rate :math:`2*L + 1`, the relationship between k and L is described by the
+    following equation:
+
+    ..math:
+
+        k = \left \lceil \log_{2}(L) \right \rceil + 1
+
+    As k does not always increase when L increases, the rate of new, unique 
+    points remains linear with respect to L.
+
+    """
+
+    def _create(self):
+        r"""Create the points in the set.
+
+        Generating nested extrema of chebyshev polynomials of the first kind.
+        """
+        # create properties for levels
```

Comment:
Make this method look like the `ClenshawCurtisPointSet` with the suggested changes. It would be good to also add some explanation for where the formula for the rule comes from.